web-ui/src/main/resources/catalog/components/common/map/esriService.js

```diff
@@ -160,6 +160,38 @@
             }
           }
           return [width, height];
+        },
+
+        /**
+         * Returns a promise resolving to the capabilities document of an ESRI Rest service.
+         * @param {String} url
+         * @return {Promise<String>} capabilities document
+         */
+        getCapabilities: function(url) {
+          var timeout = 60 * 1000;
+          var defer = $q.defer();
+          if (url) {
+            url = url + '?f=json';
+
+            if (gnUrlUtils.isValid(url)) {
+              $http.get(url, {
+                cache: true,
+                timeout: timeout
+              })
+                .success(function(data, status, headers, config) {
+                  if (!!data.mapName) {
```

Comment:
What does this check mean? This might require a comment, because it looks like you received a valid response and still decided to throw a "capabilities failed" error.

Reply:
Checks if the response has a property `mapName`, that is expected to be in the ESRI Rest service document, to avoid you configure any url like https://github.com

Reply:
Ok, then it would make sense to return another error, like "The response was not a valid ESRI Rest document" or something like that. I'm not sure this is what happens with other services, we probably just let them fail once they try to parse a non existent property or because of the content-type.